pkgs/applications/misc/blucontrol/wrapper.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, makeWrapper, ghcWithPackages, packages ? (_:[]) }:
+let
+  blucontrolEnv = ghcWithPackages (self: [ self.blucontrol ] ++ packages self);
+in
+  stdenv.mkDerivation {
+    pname = "blucontrol-with-packages";
+    version = blucontrolEnv.version;
+
+    nativeBuildInputs = [ makeWrapper ];
+
+    buildCommand = ''
+      makeWrapper ${blucontrolEnv}/bin/blucontrol $out/bin/blucontrol \
+        --prefix PATH : ${lib.makeBinPath [ blucontrolEnv ]}
+    '';
+
+    # trivial derivation
+    preferLocalBuild = true;
+    allowSubstitues = false;
```

Comment:
```suggestion
```

Reply:
> `allowSubstitutes`:
If this attribute is set to false, then Nix will always build this derivation; it will not try to substitute its outputs. This is useful for very trivial derivations (such as writeText in Nixpkgs) that are cheaper to build than to substitute from a binary cache.

Is there even any case `allowSubstitutes` should be used then?

So should it be removed from here too?
https://github.com/NixOS/nixpkgs/blob/nixos-20.09/pkgs/applications/window-managers/xmonad/wrapper.nix

I think `preferLocalBuild` can be removed then too. There are obviously no side-effects when building aside from the inputs.